app/controllers/shipit/api/locks_controller.rb

```diff
@@ -19,14 +19,29 @@ def create
         requires :reason, String, presence: true
       end
       def update
-        stack.update(lock_reason: params.reason, lock_author: current_user)
+        if stack.locked?
+          stack.update(lock_reason: params.reason, lock_author: current_user)
+        else
+          lock(params.reason, current_user)
+        end
         render_resource stack
       end
 
       def destroy
-        stack.update(lock_reason: nil, lock_author: nil)
+        unlock
         render_resource stack
       end
+
+      private
+
+      def lock(reason, user)
```

Comment:
Also this isn't the only controller that can lock the stack. Humans can set a lock via StacksController#update: https://github.com/Shopify/shipit-engine/blob/83d46c8ecb5cfc6466ff9bba95e2cabcc3861ad0/app/controllers/shipit/stacks_controller.rb#L51-L57